lib/wos_records.rb

```diff
@@ -27,6 +27,13 @@ def initialize(records: nil, encoded_records: nil)
     @doc = Nokogiri::XML(@records) { |config| config.strict.noblanks }
   end
 
+  # Select records by the database prefix in the UID
+  # @param database [String] the UID database prefix (defaults to 'WOS')
+  # @return [Nokogiri::XML::NodeSet]
+  def by_database(database = 'WOS')
+    rec_nodes.select { |rec| record_uid(rec).start_with? database }
```

Comment:
Probably should include the `:` delimiter in your matching.  I.E. `M != MEDLINE`.

Reply:
Since this method doesn't have any consumer yet, can you describe what you are expecting as a use pattern?  

If we expect consumers to iterate through both (or all) databases, might you prefer to `@return [Hash<String => Array<Nokogiri::XML::Element>>]`?  Specifically, just doing:
```ruby
rec_nodes.group_by { |rec| record_uid(rec).split(':').first }
```
Returning a each database's slice in a Hash:
```ruby
{
"WOS" => [#(Element:0x3fc69535de14 { name = "REC", children = [ #(Element:0x3fc69535c398 { name = "UID", children = [ #(Text "WOS:123")] })] })],
"MEDLINE" => [#(Element:0x3fc694cb59f4 { name = "REC", children = [ #(Element:0x3fc69536441c { name = "UID", children = [ #(Text "MEDLINE:456")] })] })]
}
```

Reply:
Good suggestions, thanks for considerate review.  Also, the immutability of the data in the WosRecords is a problem, because the NodeSet enumerable methods used in this PR return an object that still links back to the parent document.  It might be better to follow the pattern in the duplicate-related methods to first `doc.dup` and then select/filter the nodes.  Along those lines, another option is to return a new instance of the WosRecords class for each database node set.  Maybe, following your suggestion, it could be {db_key: <WosRecords>}.

Reply:
Pushed up changes, might be better.